# mRNA pipeline: HISAT2_featureCounts

This is a simple pipeline designed to perform intial stages of Illumina based mRNA sequencing.
### NOTE: **The pipeline is designed for single-end reads only!**
### NOTE: **This is a gene level, not transcript level workflow.**

## The pipeline is based on Make and covers the following steps of mRNA-sequencing workflow:
1. Initial quality control using *FastQC*.
2. Trim the reads using *Trim Galore!* with default quality cutoff (-q 20) and default Illumina adapter sequence.
3. Run FastQC with trimmed reads.
4. Map the reads to the reference genome of your choice using *HISAT2* short reads aligner with the following command:

   Command:
 
   ```hisat2 -q --rna-strandness F|R --phred33 -p 4 --known-splicesite path/to/splice_site -x path/to/index -U FASTQ -S SAM```
      
      This command can be modified as required, for more info consult HISAT2 manual: 
      https://ccb.jhu.edu/software/hisat2/manual.shtml 

5. SAM files generated by HISAT2 are converted to BAM and then indexed with *samtools*:
   Command: 
      
   ```samtools sort SAM > BAM```
   ```samtools index BAM```
   
6. Raw counts of reads mapping to features (gene level analysis) are obtained with *featureCounts*:
   Command: 

   ```featureCounts -T 4 -s 0|1|2 -a /path/to/gtf -o out_file BAM```
   
   This command can be modified as required, see the manual:
   http://bioinf.wehi.edu.au/featureCounts/

7. Collect summary statistics and intermediate quality control data using *multiqc*.
   Command: <multiqc DIR>
      

8. Intermediate SAM files are removed.

## Dependencies that need to be installed
They need to be in the PATH.
* HISAT2: https://ccb.jhu.edu/software/hisat2/index.shtml
* FastQC: https://www.bioinformatics.babraham.ac.uk/projects/fastqc/
* Trim Galore: https://www.bioinformatics.babraham.ac.uk/projects/trim_galore/
* samtools: http://samtools.sourceforge.net/
* featureCounts: http://bioinf.wehi.edu.au/featureCounts/
* multiqc: https://multiqc.info/

## Auxiliary scripts

*create_dir_tree.sh* - Run it in the project's directory to distribute the files across appropriate sub-folders.

*map_stats_bam.sh* - Run it in bam/ directory to build mapping statistics table based on *samtools flagstat* output.


## Usage
Clone the pipeline files into the Project's directory containing Single_End FASTQ files.
There has to be one FASTQ file per sample.
Execute Makefile:
Command: 
  
  ```make all```

This will run all of the steps of the pipeline.
Then run:
   
   ```bash ./scripts/create_dir_tree.sh```
    
    to create directory structure

Optional, change to *bam/* sub-folder and run 
   
   ```../scripts/map_stats_bam.sh > map_stats.txt``` 
    
   to get a simple table with total reads and mapping rates. 

# The following variables in the Makefile have to be changed to accomodate different reference genomes and library types.
path_to_splice_site := <GENOMES/Homo_sapiens/Ensembl/GRCh37/Annotation/Genes/splicesites.txt>

path_to_index := <GENOMES/Homo_sapiens/Ensembl/GRCh37/Sequence/HISAT2Index/genome>

path_to_gtf := <GENOMES/Homo_sapiens/Ensembl/GRCh37/Annotation/Genes/genes.gtf>

stranded := F

stranded_count := 1


